src/utils/workspace.ts

```diff
@@ -77,3 +77,8 @@ export function mapFilesToQuickPickItems(files: vscode.Uri[]): IAzureQuickPickIt
         };
     });
 }
+
+export function getSingleRootWorkspace(): vscode.WorkspaceFolder | undefined {
```

Comment:
How about you add `context: Partial<IDeployWizardContext>` as a parameter and then Will could re-use even more code

Reply:
Are you saying to set the context directly instead?

Reply:
Change this function to something like this:
```
return wizardContext.workspace || (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1 ? vscode.workspace.workspaceFolders[0] : undefined);
```

And call it above like this:
```
const workspaceForRecommendation: WorkspaceFolder | undefined = getSingleRootWorkspace(wizardContext);
```